src/web/View.php

```diff
@@ -643,6 +649,48 @@ public function clearJsBuffer(bool $scriptTag = true)
     }
 
     /**
+     * Registers a generic <script> code block.
+     * @param string $script the generic <script> code block to be registered
+     *
+     * @param string $key the key that identifies the generic <script> code block. If null, it will use
+     * $script as the key. If two generic <script> code blocks are registered with the same key, the latter
+     * will overwrite the former.
+     */
+    public function registerScript($script, $options = [], $key = null)
```

Comment:
There should be a `$position` argument here just like `registerJs()`, but only which only supports `POS_HEAD`, `POS_BEGIN`, and `POS_END`.